Preview Append

philippfromme opened the pull request:
![brave_LhiZ5ATZjp](https://github.com/bpmn-io/bpmn-js/assets/7633572/de791b9d-ac20-45e0-9fe0-56b998d8319d)

## Changes

* context pad
  * preview append on hover
* renderer
  * `fill`, `stroke`, `width` and `height` can be overridden
  * no other attributes can be overridden
  * public API only exposes renderers for actual BPMN elements

## Follow-Ups

* https://github.com/bpmn-io/diagram-js/issues/816

---

Closes https://github.com/bpmn-io/bpmn-js/issues/1990

philippfromme commented:
@barmac @marstamm I've added back the opacity and updated the changelog.

barmac commented on test/spec/features/append-preview/AppendPreviewSpec.js:
```diff
@@ -0,0 +1,56 @@
+    expect(canvas.getLayer('complex-preview')).to.exist;
+    expect(domQueryAll('.djs-dragger', canvas.getLayer('complex-preview'))).to.have.length(2);
+
+    // whem
```
```suggestion
    // when
```

barmac reviewed (commented):
Should we mark BpmnRenderer changes as breaking? We are removing parts of the public API.
Other than that it looks good to me. I love how it improves the UX of the editor.

marstamm reviewed (commented):
So Subprocess and Call activity are rendered with the same border width. Is this an oversight or do we have the same issues with this approach like we had when I tried to rework the renderer?

![image](https://github.com/bpmn-io/bpmn-js/assets/21984219/bf0e085d-a5a0-4e00-9e4d-1d8fc68a2f9e)

barmac commented:
> Is this an oversight or do we have the same issues with this approach like we had when I tried to rework the renderer?

Looks like a bug to me! :(

marstamm commented on lib/draw/BpmnRenderer.js:
```diff
@@ -852,445 +719,640 @@ export default function BpmnRenderer(
+    var expanded = isExpanded(element);
 
-      /* inner path */ drawRect(parentGfx, element.width, element.height, TASK_BORDER_RADIUS - 3, INNER_OUTER_DIST, innerAttrs);
+    if (!expanded) {
+      attrs = pickAttrs(attrs, [
+        'fill',
+        'stroke'
+      ]);
+    }
```
`strokeWidth` is passed as an attribute into `renderSubProcess`, but it is not picked from the attributes here

philippfromme replied:
I broke this last minute. 🙈 Makes me think I should try adding a snapshot test.

barmac commented:
Business Rule Task is broken too:

<img width="174" alt="image" src="https://github.com/bpmn-io/bpmn-js/assets/28307541/b82c74cb-cc13-4db3-913a-0c0ec1f71a4d">

Current:

<img width="173" alt="image" src="https://github.com/bpmn-io/bpmn-js/assets/28307541/54f258b7-5a69-4b31-be4a-875b4deb83c8">

philippfromme commented:
* added breaking changes to changelog
* fixed call activity stroke width
* fixed sequential marker position
* fixed business rule task icon

Ready for another review.

barmac commented:
Let's release it! 🚀
